### PR TITLE
Ansible remediation for `enable_authselect`

### LINKED ISSUE #9041: Ansible playbooks abort because the system is not configured with authselect
#### Description of problem:
During productization test "/CoreOS/scap-security-guide/Sanity/test-ansible-playbook-run ANSSI", the Ansible playbooks abort prematurely due to an error in task "Check the integrity of the current authselect profile". This happens with all 4 ANSSI profiles (minimal, intermediary, enhanced, high).

#### SCAP Security Guide Version:
scap-security-guide-0.1.63-1.git46faa00.el9.noarch

#### Operating System Version:
RHEL 9.1.0, RHEL 8.7.0

#### Steps to Reproduce:

1. ansible-playbook  --skip-tags ensure_gpgcheck_never_disabled,configure_opensc_nss_db --private-key=/root/.ssh/ssg_id_ecdsa -i "192.168.122.128," /usr/share/scap-security-guide/ansible/rhel9-playbook-anssi_bp28_minimal.yml

#### Actual Results:
The play aborts.

```

PLAY [all] *********************************************************************

TASK [Gathering Facts] *********************************************************
ok: [192.168.122.128]

TASK [Ensure aide is installed] ************************************************
changed: [192.168.122.128]

TASK [Ensure AIDE is installed] ************************************************
ok: [192.168.122.128] => (item=aide)

TASK [Build and Test AIDE Database] ********************************************
changed: [192.168.122.128]

TASK [Check whether the stock AIDE Database exists] ****************************
ok: [192.168.122.128]

TASK [Stage AIDE Database] *****************************************************
changed: [192.168.122.128]

TASK [Ensure AIDE is installed] ************************************************
ok: [192.168.122.128] => (item=aide)

TASK [Set cron package name - RedHat] ******************************************
ok: [192.168.122.128]

TASK [Set cron package name - Debian] ******************************************
skipping: [192.168.122.128]

TASK [Install cron] ************************************************************
ok: [192.168.122.128]

TASK [Configure Periodic Execution of AIDE] ************************************
changed: [192.168.122.128]

TASK [Ensure sudo is installed] ************************************************
ok: [192.168.122.128]

TASK [Ensure noexec is enabled in /etc/sudoers] ********************************
changed: [192.168.122.128]

TASK [Ensure requiretty is enabled in /etc/sudoers] ****************************
changed: [192.168.122.128]

TASK [Ensure use_pty is enabled in /etc/sudoers] *******************************
changed: [192.168.122.128]

TASK [Find /etc/sudoers.d/ files] **********************************************
ok: [192.168.122.128]

TASK [Remove lines containing !authenticate from sudoers files] ****************
ok: [192.168.122.128] => (item={'path': '/etc/sudoers'})

TASK [Find /etc/sudoers.d/ files] **********************************************
ok: [192.168.122.128]

TASK [Remove lines containing NOPASSWD from sudoers files] *********************
ok: [192.168.122.128] => (item={'path': '/etc/sudoers'})

TASK [Ensure dnf-automatic is installed] ***************************************
changed: [192.168.122.128]

TASK [Gather the package facts] ************************************************
ok: [192.168.122.128]

TASK [Ensure GPG check is globally activated] **********************************
ok: [192.168.122.128]

TASK [Gather the package facts] ************************************************
ok: [192.168.122.128]

TASK [Check stats of dnf] ******************************************************
ok: [192.168.122.128]

TASK [Check if config file of dnf is a symlink] ********************************
skipping: [192.168.122.128]

TASK [Ensure GPG check Enabled for Local Packages (dnf)] ***********************
changed: [192.168.122.128]

TASK [Read permission of GPG key directory] ************************************
ok: [192.168.122.128]

TASK [Read signatures in GPG key] **********************************************
ok: [192.168.122.128]

TASK [Set Fact - Installed GPG Fingerprints] ***********************************
ok: [192.168.122.128]

TASK [Set Fact - Valid fingerprints] *******************************************
ok: [192.168.122.128]

TASK [Import RedHat GPG key] ***************************************************
changed: [192.168.122.128]

TASK [Security patches are up to date] *****************************************
ok: [192.168.122.128]

TASK [Gather the package facts] ************************************************
ok: [192.168.122.128]

TASK [Enable timer dnf-automatic] **********************************************
changed: [192.168.122.128]

TASK [Gather the package facts] ************************************************
ok: [192.168.122.128]

TASK [Check if system relies on authselect] ************************************
ok: [192.168.122.128]

TASK [Check the integrity of the current authselect profile] *******************
fatal: [192.168.122.128]: FAILED! => {"changed": false, "cmd": ["authselect", "check"], "delta": "0:00:00.005376", "end": "2022-06-25 09:05:13.883821", "msg": "non-zero return code", "rc": 2, "start": "2022-06-25 09:05:13.878445", "stderr": "", "stderr_lines": [], "stdout": "System was not configured with authselect.", "stdout_lines": ["System was not configured with authselect."]}
...ignoring

TASK [Informative message based on the authselect integrity check result] ******
fatal: [192.168.122.128]: FAILED! => {
    "assertion": "result_authselect_check_cmd is success",
    "changed": false,
    "evaluated_to": false,
    "msg": [
        "authselect integrity check failed. Remediation aborted!",
        "This remediation could not be applied because the authselect profile is not intact.",
        "It is not recommended to manually edit the PAM files when authselect is available",
        "In cases where the default authselect profile does not cover a specific demand, a custom authselect profile is recommended."
    ]
}

PLAY RECAP *********************************************************************
192.168.122.128            : ok=35   changed=11   unreachable=0    failed=1    skipped=2    rescued=0    ignored=1   
```

#### Expected Results:
the whole Playbook is executed and completed

#### Additional Information/Debugging Steps:
no

### PR DESCRIPTION
**Pull request by mildas**
#### Description:
Ansible remediation for `enable_authselect` inspired by [Bash remediation](https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/enable_authselect/bash/shared.sh)

#### Rationale:

Fixes #9041

### CODE REVIEW

File: linux_os/guide/system/accounts/enable_authselect/ansible/shared.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# platform = multi_platform_all
+
```
Comment (ggbecker) [inline linux_os/guide/system/accounts/enable_authselect/ansible/shared.yml:2]: you will need to add all of the metadata stuff here... disruption, severity ,etc...

**Review by marcusburghardt — APPROVED** [review]
Maybe the `package` or other similar module could do the "rpm check" instead of using `command` module, but I am not sure based on the respective documentation and this would demand some deeper tests. Considering this remediation is relatively simple, this could be a possible improvement but not a problem for now.

Thanks for creating this Ansible remediation @mildas

**Comment by marcusburghardt** [comment]
Fixes https://github.com/ComplianceAsCode/content/issues/8741

**Comment by jan-cerny** [comment]
Great!

**Pull request merged**